### PR TITLE
Populate the wasi-io proposal with very preliminary content.

### PR DESCRIPTION
**Pull request by sunfishcode**
This adds very simple I/O streams and I/O arrays witx definitions,
which match read/write and pread/pwrite from wasi-filesystem, as a
starting point.

### CODE REVIEW

File: standard/io/docs.md [hunk 1]
```diff
@@ -0,0 +1,1366 @@
...
+Text file busy.
+
+- <a href="#errno.xdev" name="errno.xdev"></a> `xdev`
+Cross-device link.
+
+- <a href="#errno.notcapable" name="errno.notcapable"></a> `notcapable`
+Extension: Capabilities insufficient.
+
+## <a href="#rights" name="rights"></a> `rights`: `Record`
+File descriptor rights, determining which actions may be performed.
```
Comment (lygstate) [inline standard/io/docs.md:286]: Do we need add a rights_extended for future usage?

File: standard/io/docs.md [hunk 2]
```diff
@@ -0,0 +1,1366 @@
+# Types
+## <a href="#size" name="size"></a> `size`: `usize`
+An array size.
+
+Note: This is similar to `size_t` in POSIX.
+
+Size: 4
```
[the remark this answers is not in the document]
Comment (sunfishcode) [reply standard/io/docs.md:7]: Yes; this docs.md is auto-generated from the witx sources, where this type is declared as `(@witx usize)`, specifically to anticipate wasm64.

File: standard/io/docs.md (hunk 1 above)
Comment (sunfishcode) [reply standard/io/docs.md:286]: It's a good idea, however it turns out the rights system [isn't very useful](https://github.com/WebAssembly/WASI/issues/409#issuecomment-795494899), so it seems more likely that we'll just remove it altogether.

**Comment by sunfishcode** [comment]
Merging; this is just prelinary content; I'm planning more to come here.

**Pull request merged**